### PR TITLE
feat: add unit tests

### PR DESCRIPTION
**Pull request by ckerr**
Separates the network logic from the note parsing / comment generating logic so that the latter can be unit tested.

Adds some simple tests for detecting notes, generating notes & no-notes comments, and quoting single- and multiline notes

### CODE REVIEW

**Comment by MarshallOfSound** [comment]
Hm, #24 or #25 🤔

**Comment by nornagon** [comment]
might as well take this one since it came first :) i'll bring over my extra tests for parsing after this is merged.

**Comment by nornagon** [comment]
would be great to have `yarn run test` work :)

File: src/__tests__/note.test.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  it('can find a note', () => {
+    expect(noteUtils.findNoteInPRBody(prBodyWithNote)).toEqual('Added a memory leak.');
+  });
+  it('strips comments out of PR bodies', () => {
+    expect(noteUtils.findNoteInPRBody(prBodyWithEmbeddedComment)).toEqual('no-notes');
+  });
+});
+
+describe('comment generation', () => {
+  it('knows when to show notes ', () => {
```
Comment (nornagon) [inline src/__tests__/note.test.ts]:
```suggestion
  it('knows when to show notes', () => {
```

**Comment by nornagon** [comment]
I don't have write access to this repo but here's a patch that makes `yarn test` and `yarn test:watch` work:

```
diff --git a/jest.config.js b/jest.config.js
new file mode 100644
index 0000000..9688906
--- /dev/null
+++ b/jest.config.js
@@ -0,0 +1,17 @@
+module.exports = {
+  "roots": [
+    "<rootDir>/src"
+  ],
+  "transform": {
+    "^.+\\.tsx?$": "ts-jest"
+  },
+  "testRegex": "(/__tests__/.*|(\\.|/)(test|spec))\\.tsx?$",
+  "moduleFileExtensions": [
+    "ts",
+    "tsx",
+    "js",
+    "jsx",
+    "json",
+    "node"
+  ],
+}
diff --git a/package.json b/package.json
index 9b36781..3a75033 100644
--- a/package.json
+++ b/package.json
@@ -19,6 +19,8 @@
   },
   "scripts": {
     "build": "tsc",
+    "test": "jest",
+    "test:watch": "jest --watch",
     "lint": "tslint --project '.'",
     "start": "probot run ./lib/index.js",
     "postinstall": "tsc"
```

**Comment by ckerr** [comment]
IMO you should have access. Let's fix that.

I'm going to be in transit until tomorrow so if the spirit moves you feel free to merge this or 25 and go from there. I'm happy with whatever approach y'all want to use to move this forward.

**Pull request merged**